src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -69,7 +69,12 @@ public void RenderChildren(RazorIRNode node)
 
             public override void VisitDocument(DocumentIRNode node)
             {
+                Context.Writer.WriteLine("// <auto-generated/>");
```

Comment:
I can't think of a reason to have this not enabled. I haven't added any option to disable this behavior.